docs/api_reference/src/quantities.tex

```diff
@@ -3107,40 +3101,140 @@
+template<typename T>
+concept @\defexposconceptnc{WeaklyRegular}@ = std::@\stdconcept{copyable}@<T> && std::@\stdconcept{equality_comparable}@<T>;  // \expos
 
 template<typename T>
-concept @\defexposconceptnc{CastableNumber}@ =
-  @\exposconceptnc{CommonTypeWith}@<T, std::intmax_t> && @\exposconceptnc{ScalableNumber}@<std::common_type_t<T, std::intmax_t>>;
+concept @\defexposconceptnc{Scalar}@ = is_scalar<T>;    // \expos
 
 template<typename T>
-concept @\defexposconceptnc{Scalable}@ =
-  @\exposconceptnc{CastableNumber}@<T> ||
-  (@\exposconceptnc{CastableNumber}@<@\exposidnc{actual-value-type-t}@<T>> &&
-   @\exposconceptnc{ScalableNumber}@<T, std::common_type_t<@\exposidnc{actual-value-type-t}@<T>, std::intmax_t>>);
+concept @\defexposconceptnc{Complex}@ = is_complex<T>;  // \expos
 
 template<typename T>
-concept @\defexposconceptnc{WeaklyRegular}@ = std::@\stdconcept{copyable}@<T> && std::@\stdconcept{equality_comparable}@<T>;
+concept @\defexposconceptnc{Vector}@ = is_vector<T>;    // \expos
 
 template<typename T>
-concept @\deflibconcept{Representation}@ = (is_scalar<T> || is_complex<T> || is_vector<T> || is_tensor<T>) &&
-                         @\exposconceptnc{WeaklyRegular}@<T> && @\exposconceptnc{Scalable}@<T>;
+concept @\defexposconceptnc{Tensor}@ = is_tensor<T>;    // \expos
 
 template<typename T, quantity_character Ch>
-concept @\deflibconcept{RepresentationOf}@ =
-  @\libconcept{Representation}@<T> && ((Ch == quantity_character::scalar && is_scalar<T>) ||
+concept @\defexposconceptnc{IsOfCharacter}@ = (Ch == quantity_character::scalar && is_scalar<T>) ||  // \expos
                         (Ch == quantity_character::complex && is_complex<T>) ||
                         (Ch == quantity_character::vector && is_vector<T>) ||
-                        (Ch == quantity_character::tensor && is_tensor<T>));
+                        (Ch == quantity_character::tensor && is_tensor<T>);
+
+template<typename T>
+using @\exposidnc{scaling-factor-type-t}@ =  // \expos
+  conditional<treat_as_floating_point<T>, long double, std::intmax_t>;
+\end{itemdecl}
+
+\begin{itemdecl}
+template<typename T>
+concept @\defexposconceptnc{ScalarRepresentation}@ =  // \expos
+  @\exposconceptnc{Scalar}@<T> && @\exposconceptnc{WeaklyRegular}@<T> && requires(T a, T b, @\exposidnc{scaling-factor-type-t}@<T> f) {
+    // scaling
+    { a * f } -> Scalar;
+    { f * a } -> Scalar;
+    { a / f } -> Scalar;
+
+    // scalar operations
+    { -a } -> Scalar;
+    { a + b } -> Scalar;
+    { a - b } -> Scalar;
+    { a * b } -> Scalar;
+    { a / b } -> Scalar;
+  };
+\end{itemdecl}
+
+\pnum
+A type \tcode{T} models \libconcept{ScalarRepresentation} if
+TBD.
+
+\begin{itemdecl}
+template<typename T>
+concept @\defexposconceptnc{ComplexRepresentation}@ =  // \expos
+  @\exposconceptnc{Complex}@<T> && @\exposconceptnc{WeaklyRegular}@<T> && requires(T a, T b, @\exposidnc{scaling-factor-type-t}@<T> f) {
```

Comment:
According to our discussion, I am about to remove `is_complex` & co as well as `Complex` & co. We will detect the character based on the interfaces without the need to subscribe for it explicitly. It makes a few things easier but also others harder. To make this change work, I need to resolve many things first, so it will take a while before it is done.